src/languages/asciidoc.js

```diff
@@ -69,6 +68,36 @@ export default function(hljs) {
       begin: /\*[^\s]([^\n]+\n)+([^\n]+)\*/
     }
   ];
+  const EMPHASIS = [
+    // inline unconstrained emphasis (single line)
+    {
+      className: 'emphasis',
+      begin: /_{2}([^\n]+?)_{2}/
+    },
+    // inline unconstrained emphasis (multi-line)
+    {
+      className: 'emphasis',
+      begin: regex.concat(
+        /__/,
+        /((_(?!_)|\\[^\n]|[^_\n\\])+\n)+/,
+        /(_(?!_)|\\[^\n]|[^_\n\\])*/,
+        /__/
+      ),
+      relevance: 0
+    },
+    // inline constrained emphasis (single line)
+    {
+      className: 'emphasis',
+      // must not precede or follow a word character
+      begin: /\b_(\S|\S[^\n]*?\S)_(?!\w)/
```

Comment:
I did replace `\B` by `\b` compared to what we are doing for strong. Not sure exactly why but otherwise some tests were failing :thinking:

Reply:
You can get out the console and play with regex one at a time.. or use tools like regex101.com

Reply:
I'm using regex101.com but I can't explain why it's working with the symbol `*` but not with `_`.
Take a look:

| `\b` | `\B` |
| -- | -- |
|![nomatch-strong](https://user-images.githubusercontent.com/333276/101925881-b02f6880-3bd2-11eb-8ed8-d4a3b4926a24.png) | ![match](https://user-images.githubusercontent.com/333276/101925883-b0c7ff00-3bd2-11eb-8e0f-fdc88d522c12.png)
| ![match-empha](https://user-images.githubusercontent.com/333276/101925886-b1609580-3bd2-11eb-9166-02e026d6756b.png) | ![nomatch-empha](https://user-images.githubusercontent.com/333276/101926075-fd133f00-3bd2-11eb-8fb9-1d56399daad7.png)

Reply:
It's worth asking is `*` a word character?  Is `_`?  And there you will find your answer.